doc/NAT/SAI-NAT-API.md

```diff
@@ -0,0 +1,443 @@
+SNAT/DNAT/HARIPIN miss packets are received on a regular netdev channel to the application.
+
+- “SAI_HOSTIF_TRAP_TYPE_SNAT_MISS”
+- “SAI_HOSTIF_TRAP_TYPE_DNAT_MISS”
+- "SAI_HOSTIF_TRAP_TYPE_NAT_HAIRPIN"
+
+## 5.0 NAT Object Workflow
+ NAT or Network Address translation is used to avoid exposing the Internal or private IP addresses when the Host systems in an enterprise network or Datacenter communicate with external internet servers . The main advantage of using NAT is to reduce the usage of the Public IP addresses allocated for an enterprise or Data center.NAT feature allows the capability to change the Src IP address , Destination IP address , Src TCP/UDP port number or Destination TCP/UDP port number for an IP packet undergoing L3  routing in the switch.
+
+### 5.1 Symmetric SNAT and DNAT
+For directional symmetric flows there is always a corresponding DNAT entry for each SNAT entry. SAI APIs do not allow implicit installation of reverse direction NAT entry. SAI API MUST be invoked for both SNAT and DNAT entry to be installed.
+
+> From Zone ID: 100
+To Zone ID: 200
+VRF: None
+Packet Count: Enable
+Byte Count: Enable
+ExternalEndpoint: 65.55.42.1:1024 
+InternalEndpoint: 10.0.0.1:6000
+
+##### Step 1: Create a Source and Destination NAT Entry object:
+```sh
+sai_attribute_t nat_entry_attr[10];
+nat_entry_t snat_entry;
+
+nat_entry_attr[0].id = SAI_NAT_ENTRY_ATTR_NAT_TYPE;
+nat_entry_attr[0].value = SAI_NAT_TYPE_SOURCE_NAT;
+
+nat_entry_attr[1].id = SAI_NAT_ENTRY_ATTR_SRC_IP;
+nat_entry_attr[1].value.u32 = 65.55.42.1; //example string
+
+nat_entry_attr[2].id = SAI_NAT_ENTRY_ATTR_L4_SRC_PORT;
+nat_entry_attr[2].value.u16 = 1024;
+
+nat_entry_attr[3].id = SAI_NAT_ENTRY_ATTR_TO_ZONE;
+nat_entry_attr[3].value.u32 = 200;
+
+nat_entry_attr[4].id = SAI_NAT_ENTRY_ATTR_FROM_ZONE;
+nat_entry_attr[4].value.u32 = 100;
+
+nat_entry_attr[5].id = SAI_NAT_ENTRY_ATTR_ENABLE_PACKET_COUNT;
+nat_entry_attr[5].value.bool = true;
+
+nat_entry_attr[6].id = SAI_NAT_ENTRY_ATTR_ENABLE_BYTE_COUNT;
+nat_entry_attr[6].value.bool = true;
+
+attr_count = 7;
+
+memset(&snat_etnry, 0, sizeof(nat_etnry));
+
+snat_entry.data.key.src_ip = 10.0.0.1;
+snat_entry.data.mask.src_ip = 0xffffffff;
+snat_entry.data.key.l4_src_port = 6000;
+snat_entry.data.mask.l4_src_port = 0xffff;
+snat_entry.data.key.proto = 17;
+snat_entry.data.mask.proto = 0xff;
+
+
+create_nat_entry(&snat_entry, attr_count, nat_entry_attr);
+
+```
+
+##### Step 2: Create a Destination NAT Entry object:
+```sh
+
+sai_attribute_t nat_entry_attr[10];
+nat_entry_t dnat_entry;
+
+nat_entry_attr[0].id = SAI_NAT_ENTRY_ATTR_NAT_TYPE;
+nat_entry_attr[0].value = SAI_NAT_TYPE_DESTINATION_NAT;
+
+nat_entry_attr[1].id = SAI_NAT_ENTRY_ATTR_DST_IP;
+nat_entry_attr[1].value.u32 = 10.0.0.1; //example string
+
+nat_entry_attr[2].id = SAI_NAT_ENTRY_ATTR_L4_DST_PORT;
+nat_entry_attr[2].value.u16 = 6000;
+
+nat_entry_attr[3].id = SAI_NAT_ENTRY_ATTR_TO_ZONE;
+nat_entry_attr[3].value.u32 = 100;

```

Comment:
This attribute no longer exists, should be deleted

Reply:
removed